makepanda/makepandawrapper.py

```diff
@@ -0,0 +1,372 @@
+        group.add_argument(
+            '--use-{}'.format(pkg.lower()),
+            dest='incpkgs',
+            action='append_const',
+            const=pkg
+        )
+        group.add_argument(
+            '--no-{}'.format(pkg.lower()),
+            dest='expkgs',
+            action='append_const',
+            const=pkg
+        )
+
+    args = parser.parse_args()
+
+    # Generate a package list and store it on args
+    pkgs = set(PKG_LIST[:] if args.everything else [])
+
+    if args.incpkgs:
+        for pkg in args.incpkgs:
+            pkgs.add(pkg)
+
+    if args.expkgs:
+        for pkg in args.expkgs:
+            if pkg in pkgs:
+                pkgs.remove(pkg)
+
+    if not pkgs:
+        print(gen_usage(
+            'specify a list of packages to use or --everything to enable all packages'
+        ), file=sys.stderr)
+        sys.exit(1)
+
+    args.pkgs = pkgs
+    return args
+
+
+def opt_to_config(optlevel):
+    if optlevel == 4:
+        return 'Release'
+    if optlevel == 3:
+        return 'Standard'
+    return 'Debug'
+
+
+def gen_config(cmakedir, cli_args):
+    os.chdir(cmakedir)
+    cachepath = os.path.join(cmakedir, 'CMakeCache.txt')
+
+    args = [
+        'cmake',
+        '../',
+        '-DWANT_PYTHON_VERSION={}'.format('.'.join([str(i) for i in sys.version_info[0:2]])),
+    ]
+
+    if cli_args.arch:
+        args += [
+            '-A', 'x64'
+        ]
+    elif sys.platform == 'win32' and platform.architecture()[0] == '64bit':
+        args += [
+            '-A', 'x64'
+        ]
+
+    # Prefer Ninja if available
+    if shutil.which('ninja'):
+        args += [
+            '-G', 'Ninja'
+        ]
+
+    # Handle HAVE_* flags
+    try:
+        with open(cachepath) as cachefile:
+            found_pkgs = {
+                line.split('_')[0]
+                for line in cachefile.readlines()
+                if 'INCLUDE_DIR:PATH' in line and 'NOTFOUND' not in line
+            }
+    except FileNotFoundError:
+        found_pkgs = set()
+
+    def have_value(pkg):
+        if pkg not in cli_args.pkgs:
+            # This is easy, just disable
+            return 'OFF'
+        # Otherwise we only want to turn this ON if we've found the package
+        # before
+        return 'ON' if ARG_TO_HAVE_MAP.get(pkg, pkg) in found_pkgs else None
+
+
+    args += [
+        '-DHAVE_{}={}'.format(ARG_TO_HAVE_MAP.get(pkg, pkg), have_value(pkg))
+        for pkg in HAVE_PKGS
+        if have_value(pkg)
+    ]
+
+    # Handle BUILD_* flags
+    args += [
+        '-DBUILD_{}={}'.format(comp, 'ON' if comp in cli_args.pkgs else 'OFF')
+        for comp in BUILD_PKGS
+    ]
+
+    # Other CLI args
+    if cli_args.distributor:
+        args.append('-DPANDA_DISTRIBUTOR={}'.format(cli_args.distributor))
+
+    if cli_args.osxtarget:
+        args.append('-DCMAKE_OSX_DEPLOYMENT_TARGET={}'.format(cli_args.osxtarget))
+
+    if cli_args.static:
+        args.append('-DBUILD_SHARED_LIBS=OFF')
```

Comment:
Most settings should probably have an `else` case that sets them to the default again.  Otherwise they will "stick" if you pass them and then drop them in a future invocation.